modules/nsq/src/it/scala/com/snowplowanalytics/snowplow/enrich/nsq/test/Utils.scala

```diff
@@ -0,0 +1,135 @@
+import com.snowplowanalytics.snowplow.enrich.common.fs2.test.CollectorPayloadGen
+import com.snowplowanalytics.snowplow.enrich.nsq.test.Containers.NetworkTopology
+
+object Utils {
+
+  private implicit def logger[F[_]: Sync]: Logger[F] = Slf4jLogger.getLogger[F]
+
+  val maxBufferQueueSize = 3000
+
+  val backoffPolicy = BackoffPolicy(
+    minBackoff = 100.milliseconds,
+    maxBackoff = 10.seconds,
+    maxRetries = Some(10)
+  )
+
+  type AggregateGood = List[Event]
+  type AggregateBad = List[String]
+
+  def mkResources[F[_]: Async: ContextShift: Timer] =
+    for {
+      blocker <- Blocker[F]
+      topology <- Containers.createContainers[F](blocker)
+      sink <- Sink.init[F](
+        blocker,
+        OutNsq(
+          topology.sourceTopic,
+          "127.0.0.1",
+          topology.nsqd1.tcpPort,
+          backoffPolicy
+        )
+      )
+    } yield (blocker, topology, sink)
+
+  def generateEvents[F[_]: Sync: ContextShift: Timer](sink: ByteSink[F], goodCount: Long, badCount: Long, topology: NetworkTopology): Stream[F, Unit] =
+    CollectorPayloadGen.generate[F](goodCount, badCount)
+      .evalMap(events => sink(List(events)))
+      .onComplete(fs2.Stream.eval(Logger[F].info(s"Random data has been generated and sent to ${topology.sourceTopic}")))
+
+  def consume[F[_]: ConcurrentEffect: ContextShift](
+    blocker: Blocker,
+    refGood: Ref[F, AggregateGood],
+    refBad: Ref[F, AggregateBad],
+    topology: NetworkTopology
+  ): Stream[F, Unit] =
+    consumeGood(blocker, refGood, topology).merge(consumeBad(blocker, refBad, topology))
+
+  def consumeGood[F[_]: ConcurrentEffect: ContextShift](
+    blocker: Blocker,
+    ref: Ref[F, AggregateGood],
+    topology: NetworkTopology
+  ): Stream[F, Unit] =
+    Source.init[F](
+      blocker,
+      InNsq(
+        topology.goodDestTopic,
+        "EnrichedChannel",
+        "127.0.0.1",
+        topology.lookup2.httpPort,
+        maxBufferQueueSize,
+        backoffPolicy
+      )
+    ).evalMap(aggregateGood(_, ref))
+
+  def consumeBad[F[_]: ConcurrentEffect: ContextShift](
+    blocker: Blocker,
+    ref: Ref[F, AggregateBad],
+    topology: NetworkTopology
+  ): Stream[F, Unit] =
+    Source.init[F](
+      blocker,
+      InNsq(
+        topology.badDestTopic,
+        "BadRowsChannel",
+        "127.0.0.1",
+        topology.lookup2.httpPort,
+        maxBufferQueueSize,
+        backoffPolicy
+      )
+    ).evalMap(aggregateBad(_, ref))
+
+  def aggregateGood[F[_]: Sync](r: Record[F], ref: Ref[F, AggregateGood]): F[Unit] = {
+    for {
+      e <- Sync[F].delay(Event.parse(new String(r.data)).getOrElse(throw new RuntimeException("can't parse enriched event")))
+      _ <- r.ack
+      _ <- ref.update(updateAggregateGood(_, e))
+    } yield ()
+  }
+
+  def aggregateBad[F[_]: Sync](r: Record[F], ref: Ref[F, AggregateBad]): F[Unit] = {
+    for {
+      br <- Sync[F].delay(new String(r.data))
```

Comment:
Nit: parse as `BadRow`. Example [here](https://github.com/snowplow/enrich/blob/master/modules/kinesis/src/it/scala/com/snowplowanalytics/snowplow/enrich/kinesis/utils.scala#L94-L99) (should be moved to `common-fs2` if we reuse it here)